src/fontlibc/fontlibc.h

```diff
@@ -0,0 +1,437 @@
+ * Returns the total number of printable glyphs in the font.
+ * This can return 256; no valid font has 0 printable glyphs.
+ * @return Total number of printable glyphs
+ */
+uint24_t fontlib_GetTotalGlyphs(void);
+
+/**
+ * Returns the code point of the first printable glyph.
+ * Note that the C SDK makes char SIGNED by default, so you probably want to
+ * typecast this to unsigned char before attempting any math with it.
+ * @return First print glyph code point
+ */
+char fontlib_GetFirstGlyph(void);
+
+/**
+ * Allows you to set the code point that is recognized as being a new line code.
+ * You can set this to zero to prevent new line code processing.
+ * This defaults to 0x0A (ASCII line feed/UNIX newline)
+ * @param code_point New code point to use for newline
+ */
+void fontlib_SetNewlineCode(char code_point);
+
+/**
+ * Returns the code point that is currently recognized as being a newline.
+ * @return Current newline
+ */
+char fontlib_GetNewlineCode(void);
+
+/**
+ * Sets an alternate code point to recognize as a stop code.
+ * For example, you can set this to space to make DrawString and GetStringWidth
+ * stop processing when they reach a space.
+ * Setting this to zero will cause this to be ignored.
+ * NULL (0) will still be recognized as a stop code regardless of value.
+ * Defaults to 0.
+ * @param code_point Additional code point to recognize as a stop code.
+ */
+void fontlib_SetAlternateStopCode(char code_point);
+
+/**
+ * Returns the current alternate stop code.
+ * @return Current alternate stop code.
+ */
+char fontlib_GetAlternateStopCode(void);
+
+/**
+ * Sets the first code point considered printable.
+ * All code points before this will be considered control codes.
+ * Setting this to 0 (NULL) will NOT cause NULL to be ignored.
+ * This defaults 0x10.
+ * @param code_point First printable code point
+ */
+void fontlib_SetFirstPrintableCodePoint(char code_point);
+
+/**
+ * Returns the first code point considered printable.
+ * @return Current first printable code point
+ */
+char fontlib_GetFirstPrintableCodePoint(void);
+
+/**
+ * Returns the width of the given glyph
+ * @param codepoint Codepoint to test
+ * @return Width of glyph, 0 if invalid codepoint
+ */
+uint8_t fontlib_GetGlyphWidth(char codepoint);
+
+/**
+ * Returns the width of a string printed in the current font.
+ * Stops processing when it encounters ANY control code or a codepoint not in
+ * the current font.
+ * @param str Pointer to string
+ * @return Width of string
+ */
+uint24_t fontlib_GetStringWidth(char* str);
+
+/**
+ * Returns the width of a string printed in the current font.
+ * Stops processing when it encounters ANY control code or a codepoint not in
+ * the current font, or when max_characters have been processed.
+ * @param str Pointer to string
+ * @param max_characters Maximum number of characters to process
+ * (or 0 if not needed)
+ * @return Width of string
+ */
+uint24_t fontlib_GetStringWidthL(char* str, int24_t max_characters);
```

Comment:
Shouldn't that be `const char* str`?

Reply:
And `max_characters` might make more sense as `unsigned`.

Reply:
I mean, the getter function is signed, so that, too, is unsigned. I'd also argue that passing a negative value makes it pretty clear you never expect the limit to be reached, whereas passing a large positive value suggests you might think it could print a huge number of characters.

Reply:
Passing `-1` for a value that logically represents a count implies that you're requesting some kind of special handling, but there's neither documentation nor implementation for this. And there's no reason for there to be documentation or implementations of special handling, when simply passing a value larger than can ever practically be reached (`UINT_MAX`) achieves the same purpose but without relying on special handling.

Reply:
Fair enough. But in that case, shouldn't I change `GetCharactersRemaining` it to `bool ReachedMaxCharacters(void)`?

Also, the name implies what it returns---how many characters were left before reaching `max_characters`. If the value is negative, it tells you your previous call was to the non-L variant. I suppose change Draw/Width should accept `uint24_t` but `GetCharactersRemaining` return int24_t.

Reply:
Or alternatively, have it accept a `uint16_t` for `max_characters`. The upshot is that it explicitly ensures that the return domain fully encompasses the input domain.